atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/NameMetadataDescription.java

```diff
@@ -183,8 +185,13 @@ public String renderToJson(byte[] name) {
 
     public byte[] parseFromJson(String json, boolean allowPrefix) {
         try {
-            JSONObject obj = (JSONObject) new JSONParser().parse(json);
-            int numDefinedFields = countNumDefinedFields(obj);
+            JsonNode jsonNode = OBJECT_MAPPER.readTree(json);
+            Preconditions.checkState(jsonNode.isObject(),
+                    "Only JSON objects can be deserialized into parsed byte arrays.  Passed json was: %s",
```

Comment:
nit: I think there were two spaces after the full stop

```suggestion
                    "Only JSON objects can be deserialized into parsed byte arrays. Passed json was: %s",
```

Reply:
I wouldn't normally write this, but this is intentional given the other error messages also have this.

Reply:
Ah well, whatever.